pkg/api/norman/customization/cluster/validator.go

```diff
@@ -342,6 +346,127 @@ func (v *Validator) validateGenericEngineConfig(request *types.APIContext, spec
+// they are overwriting. If there is no previous credential such as during a create or the old credential cannot
+// be found, the auth check will succeed as long as the user can access the new credential.
+func validateAKSCredentialAuth(request *types.APIContext, credential string, prevCluster *v3.Cluster) error {
+	var accessCred mgmtclient.CloudCredential
+	credentialErr := "error accessing cloud credential"
+	if err := access.ByID(request, &mgmtSchema.Version, mgmtclient.CloudCredentialType, credential, &accessCred); err != nil {
+		return httperror.NewAPIError(httperror.NotFound, credentialErr)
+	}
+
+	if prevCluster == nil || prevCluster.Spec.AKSConfig == nil {
+		return nil
+	}
+
+	// validate the user has access to the old cloud credential before allowing them to change it
+	credential = prevCluster.Spec.AKSConfig.AzureCredentialSecret
+	if err := access.ByID(request, &mgmtSchema.Version, mgmtclient.CloudCredentialType, credential, &accessCred); err != nil {
+		if apiError, ok := err.(*httperror.APIError); ok {
+			if apiError.Code.Status == httperror.NotFound.Status {
+				// old cloud credential doesn't exist anymore, anyone can change it
+				return nil
+			}
+		}
+		return httperror.NewAPIError(httperror.NotFound, credentialErr)
+	}
+
+	return nil
+}
+
+// validateAKSKubernetesVersion checks whether a kubernetes version is provided
+func validateAKSKubernetesVersion(spec *v32.ClusterSpec, prevCluster *v3.Cluster) error {
+	clusterVersion := spec.AKSConfig.KubernetesVersion
+
+	if clusterVersion == nil {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, "cluster kubernetes version cannot be empty string")
+	}
+
+	return nil
+}
+
+// validateAKSNodePools checks whether a given NodePool version is empty or not supported.
+// More involved validation is performed in the aks-operator.
+func validateAKSNodePools(spec *v32.ClusterSpec) error {
+	nodePools := spec.AKSConfig.NodePools
+	if nodePools == nil {
+		return nil
+	}
+	if len(nodePools) == 0 {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("must have at least one nodepool"))
+	}
+
+	var errors []string
+
+	for _, np := range nodePools {
+
+		name := np.Name
+		if name == nil && *name == "" {
+			return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("nodePool Name cannot be an empty"))
+		}
+
+		version := np.OrchestratorVersion
+		if version == nil {
+			errors = append(errors, fmt.Sprintf("nodePool [%s] version cannot be empty string", *name))
+			continue
+		}
+	}
+
+	if len(errors) != 0 {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf(strings.Join(errors, ";")))
+	}
+	return nil
+}
+
+func validateAKSAccess(request *types.APIContext, aksConfig map[string]interface{}, prevCluster *v3.Cluster) error {
+	privateCluster, _ := aksConfig["privateCluster"]
+	if request.Method != http.MethodPost {
+		if privateCluster == nil {
+			privateCluster = prevCluster.Spec.AKSConfig.PrivateCluster
+		}
+	}
+
```

Comment:
What is this for? Does this function need to be finished, or can it be removed?

Reply:
This needs to be addressed.